gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/iapi/sql/dictionary/StatementTablePermission.java

```diff
@@ -134,7 +135,9 @@ public void check( LanguageConnectionContext lcc,
 	{
 		DataDictionary dd = lcc.getDataDictionary();
 	
-		if( ! hasPermissionOnTable( dd, authorizationId, forGrant))
+		if( ! hasPermissionOnTable( dd, authorizationId, forGrant) && !(getTableDescriptor(dd)
+				.getSchemaName().equalsIgnoreCase(Misc.SNAPPY_HIVE_METASTORE) && this.privType ==
```

Comment:
The existing indentation is with tabs in the entire file, so followed the same as it may not look good with some lines with space indentation and others with tabs.